crates/core_arch/src/x86/sse.rs

```diff
@@ -1251,6 +1251,25 @@ pub unsafe fn _mm_loadr_ps(p: *const f32) -> __m128 {
     simd_shuffle4(a, a, [3, 2, 1, 0])
 }
 
+/// Loads unaligned 64-bits of integer data from memory into new vector.
+///
+/// `mem_addr` does not need to be aligned on any particular boundary.
+///
+/// [Intel's documentation](https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=_mm_loadu_si64)
+#[inline]
+#[target_feature(enable = "sse")]
+#[cfg_attr(test, assert_instr(movups))]
+#[stable(feature = "simd_x86", since = "1.46.0")]
+pub unsafe fn _mm_loadu_si64(mem_addr: *const u8) -> __m128i {
+    let mut dst = _mm_setzero_si128();
+    ptr::copy_nonoverlapping(
+        mem_addr,
+        &mut dst as *mut __m128i as *mut u8,
+        8, // == 64 bits == mem::size_of::<__m128i>() / 2
+    );
+    dst
```

Comment:
The implementation could be simplified:
```rust
let val = ptr::read_unaligned(mem_addr as *const i64);
_mm_set_epi64x(val, 0)
```

Reply:
Thanks for the suggestion, but do we need `as *const i64`?

Reply:
Yes, otherwise only an `u8` would be read and not a full `i64`.

Reply:
In this particular case I think we can just change the argument type to `*const i64`. The original intrinsic uses `const void*` but we don't have that in Rust.

Reply:
But why is it `*const i64` instead of `*const u64`?

Reply:
All the other intrinsics seem to prefer using signed types over unsigned ones (e.g. `_mm_set_epi64x`). It makes no difference in practice, the bits loaded from memory are the same.

Reply:
Hmm apparently the intrinsic verifier doesn't like it so just change it back to `*mut u8` and use a cast like I suggested.